CRM/Event/BAO/Participant.php

```diff
@@ -1904,4 +1904,36 @@ public static function getSelfServiceEligibility(int $participantId, string $url
     return $details;
   }
 
+  /**
+   * Callback for hook_civicrm_pre().
+   * @param \Civi\Core\Event\PreEvent $event
+   * @throws CRM_Core_Exception
+   */
+  public static function self_hook_civicrm_pre(\Civi\Core\Event\PreEvent $event) {
+    if ($event->entity === 'Participant' && $event->action === 'create' && empty($event->params['created_id'])) {
+      // Set the "created_id" field if not already set.
+      // The created_id should always be the person that actually did the registration.
+      // That might be the first participant, but it might be someone registering someone without registering themselves.
+      // 1. Prefer logged in contact id
+      // 2. Fall back to 'registered_by_id' param.
+      // 3. Fall back to participant contact_id (for anonymous person registering themselves)
+      $event->params['created_id'] = CRM_Core_Session::getLoggedInContactID();
+      if (empty($event->params['created_id'])) {
```

Comment:
This check is redundant now - same as above

Reply:
@eileenmcnaughton No. Because we probably set it in the line above to the logged in contact ID (step 1). The other two conditions are only triggered if the created_id is still empty

Reply:
ok right